uni/lib/controller/backgroundWorkers/notifications.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:logger/logger.dart';
+import 'package:tuple/tuple.dart';
+import 'package:uni/controller/backgroundWorkers/notifications/tuition_notification.dart';
+import 'package:uni/controller/local_storage/app_shared_preferences.dart';
+import 'package:uni/controller/local_storage/notification_timeout_storage.dart';
+import 'package:uni/controller/networking/network_router.dart';
+import 'package:uni/model/entities/session.dart';
+import 'package:workmanager/workmanager.dart';
+
+///
+/// Stores all notifications that will be checked and displayed in the background. 
+/// Add your custom notification here, because it will NOT be added at runtime.
+/// (due to background worker limitations).
+/// 
+Map<Type, Notification Function()> notificationMap = {
+  TuitionNotitification:() => TuitionNotitification()
+};
+
+
+
+
+abstract class Notification{
+  
+  String uniqueID;
+  Duration timeout;
+
+  Notification(this.uniqueID, this.timeout);
+
+  Future<Tuple2<String, String>> buildNotificationContent(Session session);
+
+  Future<bool> checkConditionToDisplay(Session session);
+
+  void displayNotification(Tuple2<String, String> content, FlutterLocalNotificationsPlugin localNotificationsPlugin);
+
+  Future<void> displayNotificationIfPossible(Session session, FlutterLocalNotificationsPlugin localNotificationsPlugin) async{
+    final bool test = await checkConditionToDisplay(session);
+    Logger().d(test);
+    if(test){
+      displayNotification(await buildNotificationContent(session), localNotificationsPlugin);
+    }
+  }
+}
+
+class NotificationManager{
+
+  static const Duration startDelay = Duration(seconds: 15);
+
+  static final FlutterLocalNotificationsPlugin _localNotificationsPlugin = FlutterLocalNotificationsPlugin();
+
+
+
+  static Future<void> tryRunAll() async{
```

Comment:
I suggested it when we were first implementing this, but we should swap this methods' name with something more meaningful like`updateAndDisplayNotifications` || `updateAndTriggerNotifications`.